File: jpms-java/src/test/java/example/jpms_java/UpdateTest.java
package example.jpms_java;

import static org.junit.jupiter.api.Assertions.assertNotNull;

import example.jpms_java.entity.Employee;
import example.jpms_java.entity.JobType;
import example.jpms_java.repository.EmployeeRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class UpdateTest {

  private final EmployeeRepository repository;

  UpdateTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testUpdate() {
    Employee employee = repository.selectById(1);
    assertNotNull(employee);
    employee.setName("hoge");
    employee.setJobType(JobType.PRESIDENT);
    repository.update(employee);
  }

  @Test
  public void testUpdateByNativeSql() {
    Employee employee = repository.selectById(1);
    assertNotNull(employee);
    employee.setName("hoge");
    employee.setJobType(JobType.PRESIDENT);
    repository.updateByNativeSql(employee);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl_style_kotlin/UpdateTest.kt
package example.dsl_style_kotlin

import example.dsl_style_kotlin.entity.JobType
import example.dsl_style_kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class UpdateTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testUpdate() {
        val employee = repository.selectById(1)
        assertNotNull(employee)
        employee.name = "hoge"
        employee.jobType = JobType.PRESIDENT
        repository.update(employee)
    }

    @Test
    fun testUpdateByNativeSql() {
        val employee = repository.selectById(1)
        assertNotNull(employee)
        employee.name = "hoge"
        employee.jobType = JobType.PRESIDENT
        repository.updateByNativeSql(employee)
    }
}
